src/compiler.ts

```diff
@@ -367,6 +367,8 @@ export class Compiler extends DiagnosticEmitter {
   virtualCalls: Set<Function> = new Set();
   /** Elements currently undergoing compilation. */
   pendingElements: Set<Element> = new Set();
+  /** Classes that are already added as globals */
+  globalClasses: Set<Class> = new Set();
```

Comment:
Comment and name here are somewhat ambiguous. Can easily be misinterpreted as if this has something to do with the global scope or elements `@global` to the program. What we are tracking here in particular is "classes, that are module exports, already processed". Perhaps one way to model this in a more general way is to make the map contain all the module exports already processed, including functions and globals, and name the set `doneModuleExports` of type `Element`? While we really only check classes currently, we'd still get a somewhat complete concept out of it this way :)